component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/totp/TOTPAuthenticator.java

```diff
@@ -739,6 +746,10 @@ private void handleTotpVerificationFail(AuthenticationContext context) throws Au
 			updatedClaims
 					.put(TOTPAuthenticatorConstants.TOTP_FAILED_ATTEMPTS_CLAIM, String.valueOf(currentAttempts + 1));
 			setUserClaimValues(authenticatedUser, username, updatedClaims);
+            IdentityErrorMsgContext customErrorMessageContext =
+                    new IdentityErrorMsgContext(UserCoreConstants.ErrorCode.INVALID_CREDENTIAL, currentAttempts + 1,
+                            maxAttempts);
+            IdentityUtil.setIdentityErrorMsg(customErrorMessageContext);
```

Comment:
As discussed offline if the account locked is enabled and account is not locked only this will get hit

Reply:
Yes, account lock enabled when totp attempt failure exceed is enabled.
When the failed attempts not exceed the defined limit, this logic get executed. It will update the failing attempt count. With the new logic, the failing attempt count and the totp failure is communicated back for the processing.